src/Platform/Microsoft.Testing.Platform/Services/CurrentTestApplicationModuleInfo.cs

```diff
@@ -52,11 +52,27 @@ public string GetCurrentTestApplicationFullPath()
 
         if (_runtimeFeature.IsDynamicCodeSupported)
         {
+            // Application is running non-native, this is safe to call.
```

Comment:
If you support single file, this is not safe to call. This could be a single file app (`PublishSingleFile`) that returns the same thing from `Assembly.GetEntryAssembly()?.Location` as native AOT.

The better way to check for single file is to compare the `moduleName` you got here with an empty string instead of trying to approximate with IsDynamicCodeSupported.

I think this would not just fix PublishSingleFile, but also fix your bug.

Pretty much every time you think you want to detect Native AOT, you need to step back.

Reply:
Used this approach, and it works.